### PR TITLE
Learn the predictor

### PR DESCRIPTION
**Pull request by turian**
Unfortunately, we skip dcase because it doesn't have a validation set. https://github.com/neuralaudio/hear-eval-kit/issues/94

### CODE REVIEW

File: heareval/predictions/task_predictions.py [hunk 1]
```diff
@@ -36,30 +49,65 @@ def __init__(self, nfeatures: int, nlabels: int, prediction_type: str):
...
 
 
+class PredictionModel(pl.LightningModule):
+    def __init__(self, nfeatures: int, nlabels: int, prediction_type: str):
+        super().__init__()
+
+        self.predictor = RandomProjectionPrediction(nfeatures, nlabels, prediction_type)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.predictor(x)
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:76]: So is the activation function only applied during testing?

File: heareval/predictions/task_predictions.py [hunk 2]
```diff
@@ -257,22 +304,43 @@ def label_vocab_nlabels(embedding_path: Path) -> Tuple[pd.DataFrame, int]:
...
         )
     elif metadata["embedding_type"] == "event":
-        predictor = RandomProjectionPrediction(
+        predictor = PredictionModel(
             timestamp_embedding_size, nlabels, metadata["prediction_type"]
         )
     else:
         raise ValueError(f"embedding_type {metadata['embedding_type']} unknown.")
+    # train on CPU
+    # TODO: FIXME
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:339]: What's broken?

**Review by jorshi — APPROVED** [review]
Looks cool!

File: heareval/predictions/task_predictions.py (hunk 1 above)
Comment (turian) [reply heareval/predictions/task_predictions.py:76]: yes, but it is in forward too

File: heareval/predictions/task_predictions.py (hunk 2 above)
Comment (turian) [reply heareval/predictions/task_predictions.py:339]: just that we want to specify more parameters here, like device

**Pull request merged**